src/components/Modals/Send/hooks/useSendDetails/useSendDetails.test.tsx

```diff
@@ -0,0 +1,327 @@
+      const { result } = setup({
+        assetBalance: balances.ethereum,
+        accountBalances: getEthAccountBalances()
+      })
+      expect(result.current.amountFieldError).toBe(null)
+      expect(result.current.balancesLoading).toBe(false)
+      expect(result.current.fieldName).toBe('fiat.amount')
+      expect(result.current.loading).toBe(false)
+    })
+  })
+
+  it('toggles the input field', async () => {
+    return await act(async () => {
+      const { waitForValueToChange, result } = setup({
+        assetBalance: balances.ethereum,
+        accountBalances: getEthAccountBalances()
+      })
+      expect(result.current.fieldName).toBe('fiat.amount')
+      act(() => {
+        result.current.toggleCurrency()
+      })
+      await waitForValueToChange(() => result.current.fieldName)
+      expect(result.current.fieldName).toBe('crypto.amount')
+    })
+  })
+
+  it('toggles the amount input error to the fiat.amount/crypto.amount field', async () => {
+    return await act(async () => {
+      let setError = jest.fn()
+      const { waitForValueToChange, result } = setup({
+        assetBalance: balances.ethereum,
+        accountBalances: getEthAccountBalances(),
+        formErrors: {
+          fiat: { amount: { message: 'common.insufficientFunds' } }
+        },
+        setError
+      })
+
+      act(() => {
+        result.current.toggleCurrency()
+      })
+
+      await waitForValueToChange(() => result.current.fieldName)
+
+      expect(result.current.fieldName).toBe('crypto.amount')
+      expect(setError).toHaveBeenCalledWith('crypto.amount', {
+        message: 'common.insufficientFunds'
+      })
+    })
+  })
+
+  it('handles input change on fiat.amount', async () => {
+    const setValue = jest.fn()
+    await act(async () => {
+      const { waitForValueToChange, result } = setup({
+        assetBalance: balances.ethereum,
+        accountBalances: getEthAccountBalances(),
+        setValue
+      })
+      // Field is set to fiat.amount
+      expect(result.current.fieldName).toBe('fiat.amount')
+
+      // Set fiat amount
+      act(() => {
+        result.current.handleInputChange('3500')
+        expect(setValue).toHaveBeenCalledWith('crypto.amount', '1')
+        setValue.mockClear()
+
+        result.current.handleInputChange('0')
+        expect(setValue).toHaveBeenCalledWith('crypto.amount', '0')
+        setValue.mockClear()
+      })
+
+      // toggle field to crypto.amount
+      act(() => {
+        result.current.toggleCurrency()
+      })
+      await waitForValueToChange(() => result.current.fieldName)
+      expect(result.current.fieldName).toBe('crypto.amount')
+
+      // Set crypto amount
+      act(() => {
+        result.current.handleInputChange('1')
+        expect(setValue).toHaveBeenCalledWith('fiat.amount', '3500')
+        setValue.mockClear()
+      })
+    })
+  })
+
+  it('validates the fiat amount', async () => {
+    return await act(async () => {
```

Comment:
is returning here help with cleaning up some lingering act warnings?

Reply:
Yes

Reply:
Cool I have not seen this yet. I will for sure be using in the future.